squiggy/lib/db_util.py

```diff
@@ -40,42 +40,45 @@ def select_column(sql):
 @contextmanager
 def advisory_lock(lock_id):
     if not lock_id:
-        yield
+        yield False
         return
     with db.engine.connect() as lock_connection:
         # Detaching will protect the server-session PID from action in the yield block.
         # Because detached connections are permanently removed from the pool, it also
         # ensures that the lock will be released when the connection is closed.
         lock_connection.detach()
-        locked = try_advisory_lock(lock_connection, lock_id)
-        if locked:
-            try:
-                yield
-            except Exception as e:
-                app.logger.exception(e)
-                raise e
-            finally:
-                # Explicit unlocking should not be necessary, but the logging might be useful.
-                advisory_unlock(lock_connection, lock_id)
+
+        if not try_advisory_lock(lock_connection, lock_id):
+            yield False
+            return
+
+        try:
+            yield True
+        except Exception as e:
+            logger.exception(e)
+            raise e
+        finally:
+            # Explicit unlocking should not be necessary, but the logging might be useful.
+            advisory_unlock(lock_connection, lock_id)
 
 
 def try_advisory_lock(connection, lock_id):
     result = connection.execute(f'SELECT pg_try_advisory_lock({lock_id}) as locked, pg_backend_pid() as pid')
     (locked, pid) = next(result)
     if locked:
-        app.logger.info(f'Granted advisory lock {lock_id} for PID {pid}')
+        logger.info(f'Granted advisory lock {lock_id} for PID {pid}')
```

Comment:
Since the advisory lock code is called from background threads, we want logging in the appropriate thread-specific logfile rather than the main app log.